bootstrap/src/basis/intnb.ml

```diff
@@ -105,15 +105,14 @@ module Make_common (T : I_common) : S_common with type t := usize = struct
       assert (narrow t0 = t0);
       assert (narrow t1 = t1);
       let rel = match T.signed || (Sys.int_size > T.num_bits) with
-        | true -> compare t0 t1
-        | false -> compare (t0 + min_int) (t1 + min_int)
+        | true -> Stdlib.Int.compare t0 t1
+        | false -> Stdlib.Int.compare (t0 + min_int) (t1 + min_int)
       in
-      if rel < 0 then
-        Cmp.Lt
-      else if rel = 0 then
-        Cmp.Eq
-      else
-        Cmp.Gt
+      match rel with
+      | (-1) -> Cmp.Lt
+      | 0 -> Cmp.Eq
+      | 1 -> Cmp.Gt
+      | _ -> if rel < 0 then Cmp.Lt else Cmp.Gt
```

Comment:
Looks like Stdlib.Int.compare won't do this, but it's not documented. Feels sketchy, but would more optimal code be generated with a `not_reached ()`?

Reply:
I waffled on the `not_reached` approach, and eventually decided against it both because it didn't seem to help performance, and because the docs didn't promise results in {-1,0,1}.

Reply:
Oh nice, that compare function you linked to is a great way to implement comparison, but I didn't even try it because I expected the compiler to poorly optimize it.  I'll give it a try!